src/content/blog/2023/03/22/react-labs-what-we-have-been-working-on-march-2023.md

```diff
@@ -0,0 +1,103 @@
+
+## Asset Loading {/*asset-loading*/}
+
+[Suspense](/reference/react/Suspense) lets you specify what to display on the screen while the data or code for your components is still being loaded. This lets your users progressively see more content while the page is loading, and later during router navigations that load more data and code. However, from the user's perspective, data loading and rendering do not tell the whole story when considering whether new content is ready. By default, browsers load stylesheets, fonts, and images independently, which can lead to UI jumps and consecutive layout shifts.
+
+We're working to fully integrate Suspense with the loading lifecycle of stylesheets, fonts, and images, so that React takes them into account to determine whether the content is ready to be displayed. Without any change to the way you author your React components, updates will behave in a more coherent and pleasing manner. As an optimization, we will also provide a manual way to preload assets like fonts directly from components.
+
+We currently implementing these features and will have more to share soon.
+
+## Document Metadata {/*document-metadata*/}
+
+Different pages and screens in your app may have different metadata like the `<title>` tag, description, and other `<meta>` tags specific to this screen. From the maintenance perspective, it's more scalable to keep this information close to the React component for that page or screen. However, the HTML tags for this metadata need to be in the document `<head>` which is typically rendered in a component at the very root of your app.
+
+Today, people solve this problem with one of the two techniques.
+
+One technique is to render a special third-party component that moves `<title>`, `<meta>`, and other tags inside it into the document `<head>`. This works for major browsers but there are many clients which do not run client-side JavaScript, such as Open Graph parsers, and so this technique is not universally suitable.
+
+Another technique is to server-render the page in two parts. First, the main content is rendered and all such tags are collected. Then, the `<head>` is rendered with these tags. Finally, the `<head>` and the main content are sent to the browser. This approach works, but it prevents you from taking advantage of the [React 18's Streaming Server Renderer](/reference/react-dom/server/renderToReadableStream) because you'd have to wait for all content to render you can send the `<head>`.
+
+This is why we've created a solution that addresses all of these problems in a unified way. In short, tags like `<meta>`, `<link>`, and `<title>` anywhere in your component tree will be automatically deduplicated by React, preloaded during streaming, and inserted into the `<head>` automatically as they are discovered in the tree. This would work the same way in all environments, including fully client-side code, SSR, and in the future, RSC.
```

Comment:
we no longer deduplicate document metadata, you still need to ensure you only render one title for instance otherwise there will be more than one title in the head.

Additionally most link types are not preloadable. We do selectively preload stylesheets that are not otherwise going to be hoisted but that's a very minor use case.

Finally, when streaming we don't guarantee every hoistable tag makes it into the head on the client b/c it's not actually all that useful. OG parsers can actually read the tags in the body for instance. Only stylesheets always hoist b/c their order is critical but then this isn't really document metadata.

Reply:
```suggestion
This is why we've created a solution that addresses all of these problems in a unified way. In short, tags like `<meta>`, `<link>`, and `<title>` anywhere in your component tree will be hoisted by React to the right place in the Document, preloaded if necessary during streaming, automatically as they are discovered in the tree. This will work the same way in all environments, including fully client-side code, SSR, and in the future, RSC.
```